src/spss/readstat_sav_date.c

```diff
@@ -0,0 +1,81 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <math.h>
+
+static inline int is_leap(int year) {
+    return ((year % 4 == 0 && year % 100 != 0) || year % 400 ==0);
+}
+
+double readstat_sav_date_parse(const char *s, char **dest) {
+    // A SPSS date stored as the number of seconds since the start of the Gregorian calendar (midnight, Oct 14, 1582)
+    // Through the C interface in savReaderWriter I've verifed that leap seconds is ignored
+    int daysPerMonth[] =     {31,28,31,30,31,30,31,31,30,31,30,31};
+    int daysPerMonthLeap[] = {31,29,31,30,31,30,31,31,30,31,30,31};
+    int year, month, day;
+    if (strlen(s) == 0) {
+        *dest = (char*) s;
+        return 0;
+    }
+    int ret = sscanf(s, "%d-%d-%d", &year, &month, &day);
+    month--;
+    if (month < 0 || month > 11 || ret!=3) {
+        *dest = (char*)s;
+        return 0;
+    }
+    int maxdays = (is_leap(year) ? daysPerMonthLeap : daysPerMonth)[month]; 
+    if (day < 1 || day > maxdays) {
+        *dest =(char*)s;
+        return 0;
+    } else {
+        int days = 0;
+
+        for (int i=1582; i<year; i++) {
+            days += is_leap(i) ? 366 : 365;
+        }
+       
+        for (int m=0; m<month; m++) {
+            days += is_leap(year) ? daysPerMonthLeap[m] : daysPerMonth[m];
+        }
+
+        days += day-1;
+        char buf[1024];
+        *dest = (char*)s + snprintf(buf, sizeof(buf)-1, "%d-%d-%d", year, month+1, day); 
```

Comment:
Just FYI generally you'll want `snprintf(buf, sizeof(buf), ...)`, no need for `-1`. In this case it doesn't really matter though.